opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/SpanDecorator.java

```diff
@@ -77,16 +90,32 @@ public void onResponse(HttpServletRequest httpServletRequest, HttpServletRespons
         }
 
         @Override
-        public void onError(HttpServletRequest httpServletRequest, Throwable exception, Span span) {
+        public void onError(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                            Throwable exception, Span span) {
             Tags.ERROR.set(span, Boolean.TRUE);
             span.log(logsForException(exception));
+
+            if (httpServletResponse.getStatus() == HttpServletResponse.SC_OK) {
+                // exception is thrown in filter chain, but status code is incorrect
+                Tags.HTTP_STATUS.set(span, 500);
+            }
+        }
+
+        @Override
+        public void onTimeout(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                              long timeout,
+                              Span span) {
+            span.setTag("timeout", Boolean.TRUE);
```

Comment:
Just wondering whether this should be recorded as an error? but then log error type as timeout? Either way it probably needs to be discussed as part of the standard tags/logs in the spec.